positron/components/ModuleLoader.jsm

```diff
@@ -182,18 +197,41 @@ function ModuleLoader(processType) {
     }
   };
 
-  // Finish defining the properties of the global.process object.
-  // We do this by importing a module with our own require() function,
-  // which gives it access to that require() function, so it can use it
-  // to load native bindings.  Perhaps it would be better to populate
-  // the process object entirely within this ModuleLoader constructor.
+  // The `process` global is complicated.  It's implemented as a module
+  // that uses `require` to import other modules, but it also needs to exist
+  // before those modules are evaluated, since it's a global that's supposed
+  // to be available in them.
+  //
+  // So we declare it first here as a stub, then require the module to finish
+  // initializing it.
+  //
+  this.global.process = {
+    type: processType,
+  };
   this.require({}, 'resource:///modules/gecko/process.js');
+
+  this.global.Buffer = this.require({}, 'resource:///modules/node/buffer.js').Buffer;
+
+  // Require the Electron init.js script for the given process type.
+  //
+  // We only do this for renderer processes for now.  For browser processes,
+  // we instead require the one browser process module that renderer/init.js
+  // depends on having already been required, browser/rpc-server.js.
+  //
+  // Eventually, we should get browser/init.js working and require it here,
+  // at which point it'll require browser/rpc-server.js for us.
+  //
+  if (processType === 'renderer') {
+    this.require({}, `resource:///modules/${processType}/init.js`);
```

Comment:
> Nit: this one can only ever be "renderer", so it's unnecessary to make this a template string. I'd suggest just making it a normal string.

That's true right now, but eventually I hope to be able to use the same line to load both *browser* and *renderer* init.js scripts. In the meantime, I've hardcoded the values in 7d8976f.